doc/source/examples_source/00-basic/07-volume_render.py

```diff
@@ -0,0 +1,181 @@
+q_file = f"{session.cei_home}/ensight{session.cei_suffix}/data/plot3d/shuttle.q"
+session.load_data(
+    data_file=xyz_file,
+    result_file=q_file,
+    file_format="PLOT3D",
+    representation="3D_feature_2D_full",
+)
+session.show("image", width=800, height=600)
+
+###############################################################################
+# The PLOT3D reader only reads the volume by default. Now, extract a
+#  particular IJK range for the surface of the shuttle
+# ------------------------------------------------------------------
+
+# LPARTS are loadable parts in EnSight. The idea here is to take the
+# only structured part available and build after it a 2D surface, that
+# is the body surface. This is accomplished just selecting the max
+# values of the I,J,K directions and then loading the LPART
+lpart = session.ensight.objs.core.CURRENTCASE[0].LPARTS[0]
+lpart.NODEMAX = [53, 63, 1]
+lpart.load()
+
+session.show("image", width=800, height=600)
+
+##############################################################################
+# Volume Render Flow
+# ------------------------------------------------------------------
+#
+# .. image:: /_static/07_volume_render_1.png
+#
+#
+# Make the shuttle surface an uninteresting, uniform color grey
+#
+
+# PARTS[1] is the body surface, just created
+session.ensight.objs.core.PARTS[1].COLORBYRGB = [0.6, 0.6, 0.6]
+session.ensight.view.highlight_parts("OFF")
+
+#
+# Color the 3D flow field by the variable that we
+#  wish to visualize using Volume Rendering, Mach
+#
+
+# PARTS[0] is the 3D flow part
+session.ensight.objs.core.PARTS[0].COLORBYPALETTE = "Mach"
+session.ensight.function.palette("Mach")
+session.ensight.function.modify_begin()
+session.ensight.function.restore_predefinedpal("use_new_levels", "Magma")
+session.ensight.function.modify_end()
+
+#
+#  Volume Rendering, in EnSight, is accomplished by subdividing
+#   the flow field into a finite number of hex elements and then
+#   doing the rendering of the variable in each box according to
+#   the normalized variable value and the opacity ('alpha') for
+#   ranges of values set using function points and the magnitude
+#
+
+session.ensight.part.select_begin(1)
+session.ensight.clip.begin()
+session.ensight.part.colorby_palette("Mach")
+session.ensight.clip.domain("volume")
+session.ensight.clip.tool("xyz_box")
+session.ensight.clip.sample_step(128, 128, 128)
+session.ensight.clip.origin(-0.497170001, 1.27204275, -0.879772604)
+#
+session.ensight.clip.axis("x", 0, 0, 1)
+session.ensight.clip.axis("y", 0, -1, 0)
+session.ensight.clip.axis("z", 1, 0, 0)
+
+session.ensight.clip.length(2.24244738, 1.39883041, 3.60410643)
+session.ensight.clip.end()
+session.ensight.clip.create()
+session.ensight.legend.width(0.0244169608)
+session.ensight.legend.height(0.662014842)
+session.ensight.function.palette("Mach")
+
+
+session.ensight.function.point(0, "alpha", 0.000000, 0.000000)
```

Comment:
Not a big fan of this API at all.   Yes, it is what we have at the moment, but another "improvement" candidate.